core/templates/dev/head/pages/admin/misc_tab/AdminMiscTabDirective.js

```diff
@@ -76,6 +79,55 @@ oppia.directive('adminMiscTab', [
         $scope.downloadTopicSimilaritiesFile = function() {
           $window.location.href = ADMIN_TOPICS_CSV_DOWNLOAD_HANDLER_URL;
         };
+
+        var setDataExtractionQueryStatusMessage = function(message) {
+          $scope.showDataExtractionQueryStatus = true;
+          $scope.dataExtractionQueryStatusMessage = message;
+        };
+
+        $scope.submitQuery = function() {
+          var STATUS_PENDING = (
+            'Data extraction query has been submitted. Please wait.');
+          var STATUS_FINISHED = 'Loading the extracted data ...';
+          var STATUS_FAILED = 'Error, ';
+
+          var data = {
+            exp_id: $scope.exp_id,
+            exp_version: $scope.exp_version,
+            state_name: $scope.state_name,
+            num_answers: $scope.num_answers
+          };
+
+          setDataExtractionQueryStatusMessage(STATUS_PENDING);
+
+          $http.post(DATA_EXTRACTION_QUERY_HANDLER_URL, data).then(
+            function(response) {
+              var data = response.data;
+              if (response.data.success) {
+                setDataExtractionQueryStatusMessage(STATUS_FINISHED);
+                var extractedAnswers = JSON.stringify(data.extracted_data);
+                var dataURL = (
+                  'data:application/json;base64,' + btoa(extractedAnswers));
+                var downloadURL = (
+                  '<a href="' + dataURL +
+                  '" download="data.json"> download </a>');
+                $window.location.href = (
+                  'data:text/html;base64,' + btoa(downloadURL));
+                return;
+              }
+              setDataExtractionQueryStatusMessage(STATUS_FAILED + data.message);
+            }, function(response) {
+            setDataExtractionQueryStatusMessage(
```

Comment:
Weird. That doesn't make sense to me, actually, but we can leave it.

One thing you could try, if you like, is to indent line 118 by 2 and move the ")" to the next line so that the brackets match.